doc/design/test-dandisets-1.md

```diff
@@ -0,0 +1,102 @@
+# Handling "Test" Dandisets
+
+As we introduce people to the archive, and as we ourselves work on the archive,
+we often create "test dandisets" that are (1) devoid of assets and/or (2) are
+only meant for exploration and development.
+
+Due to the relative ease of creating a dandiset (to check out the system),
+compared to curating a dandiset for actual work, the production instance has
+more such test dandisets in it than "real" ones. This in turn degrades the
+perception of the archive as a serious place to host data, and so we need some
+way to (1) discourage the creation and abandonment of exploratory dandisets; (2)
+mark dev dandisets in some way to prevent them from appearing in the main
+listing; and (3) clean up the archive from such abandoned dandisets.
+
+## Proposals
+
+### "Test" flag
+
+This proposes to add a `testing` flag to all dandisets. At creation time, a
+dandiset can optionally have this flag set to prevent it from showing up in
```

Comment:
I would also suggest a pre-test flag step that simply points users to the staging archive.

Reply:
Good idea. Check out cc13895fc8eda7eb970ba5983d2199b80914f095.

I am, however, starting to worry about the UX here. For one thing, calling it a "staging" server may start to be troublesome if the intent is for end users to see it as a playground or testbed of sorts. We've already had one confused user ask if stuff they upload there is actually being "staged" for publishing to the production server. A change in nomenclature may be in order if we want to push this aspect of our staging server.

Otherwise, perhaps marking a dandiset as "test" should simply prevent the user from publishing it, and would forcibly delete/GC that dandiset after a set time (7 days)? Then we could retain the staging server in the devops meaning of the term.